addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -688,6 +688,69 @@ publicWidget.registry.CarouselBootstrapUpgradeFix = publicWidget.Widget.extend({
     },
 });
 
+publicWidget.registry.CarouselSingleScrollMode = publicWidget.Widget.extend({
+    selector: ".carousel.o_slide_single",
+    disabledInEditableMode: false,
+    events: {
+        'slide.bs.carousel': '_onSlide',
+        'slid.bs.carousel': '_onSlideEnd',
+    },
+
+    /**
+     * Allow the carousel to cycle otherwise it would reach an end
+     * This method is called after when the carousel is sliding
+     *
+     * @private
+     */
+    _onSlide(e) {
+        const $carousel = $(e.target);
+        const {$carouselItems, totalItems, itemsPerSlide} = this._getCarouselData($carousel);
+
+        const $e = $(e.relatedTarget);
+        const idx = $e.index();
+
+        // Check if we're near the end of the carousel and need to append items
+        if (idx >= totalItems - itemsPerSlide + (idx === 1)) {
+            const it = itemsPerSlide - (totalItems - idx) + 1;
+            for (let i = 0; i < it; i++) {
+                $carouselItems.eq(i).appendTo($carousel.find(".carousel-inner"));
+            }
+        }
+    },
+
```

Comment:
Allow the carousel to loop, so it doesn't stop at an empty slide before returning to the first one.